Woodall_Robert/Algorithms/algotemplate.htm

```diff
@@ -0,0 +1,56 @@
+<html>
+<head>
+  <title></title>
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.0/jquery.min.js"></script>
+  <script type="text/javascript">
+  
+  $(document).ready(function(){
+    // zip arrays into map (p.55)
+    function arraysToMap(arr1, arr2) {
+      var dict = {};
+
+      for (var i = 0; i < arr1.length && i < arr2.length; i++) {
+        dict[arr1[i]] = arr2[i];
+      }
+
+      // print return value for verification
+      console.log("\narraysToMap results...");
+      for (key in dict) {
+        console.log(key + ": " + dict[key]);
+      }
+
+      return dict;
+    }
+
+    // invert hash (p.55)
+    function invertHash(hash) {
+      invertedHash = {};
+
+      for (key in hash) {
+        invertedHash[hash[key]] = key;
+      }
+
+      // print return value for verification
+      console.log("\ninvertHash results...")
+      for (key in invertedHash) {
+        console.log(key + ": " + invertedHash[key]);
+      }
+
+      return invertedHash;
+    }
+
+    // function tests
+    arraysToMap(["abc", 3, "yo"], [42, "wassup", true]);
+    invertHash({"name": "zaphod", "charm": "high", "morals": "dicey"});
```

Comment:
Good job. You should be able to see the results in the console. Since you're using 'return' at the end of your functions, no need to manually print out key:value pairs. Just do something like:

```
result = arraysToMap(["abc", 3, "yo"], [42, "wassup", true]);
console.log(result)
result = invertHash({"name": "zaphod", "charm": "high", "morals": "dicey"});
console.log(result)
```

Also, rename your template file to the name of the algorithm, since every day you'll be adding new algos.